app.ts

```diff
@@ -0,0 +1,319 @@
+		// Gets the created rows.
+		let finalInfoDataRow = document.getElementById("info-row-" + columnData[0]);
+		if (finalInfoDataRow !== null) {
+			// Loops through
+			for (let x of columnData) {
+				let infoData = `<p class="info-row-data">${x}</p>`;
+				finalInfoDataRow.innerHTML += infoData;
+			}
+		}
+	}
+}
+
+function resizeScreenData() {
+	let toNumber: number;
+	let recordNav: HTMLElement | null = document.getElementById("record-navigation-container");
+	let nextBtn: HTMLElement | null = document.getElementById("next-records-btn");
+	let previousBtn: HTMLElement | null = document.getElementById("previous-records-btn");
+	let navBtns = document.getElementById("navigation-btns");
+	if (recordNav !== null) {
+		if (recordNav.contains(navBtns as HTMLDivElement)) {
+			let numberOfRows = Math.floor(window.innerHeight / 50);
+
+			(nextBtn as HTMLButtonElement).disabled = false;
+			(previousBtn as HTMLButtonElement).disabled = false;
+
+			let finalRecord = recordNumberTotal - 1;
+
+			if (fromNumber + numberOfRows >= finalRecord) {
+				fromNumber = finalRecord - numberOfRows;
+				(nextBtn as HTMLButtonElement).disabled = true;
+				(previousBtn as HTMLButtonElement).disabled = false;
+			} else if (fromNumber <= 0) {
+				(nextBtn as HTMLButtonElement).disabled = false;
+				(previousBtn as HTMLButtonElement).disabled = true;
+				fromNumber = 0;
+			}
+
+			toNumber = fromNumber + numberOfRows;
+
+			getRecords(fromNumber, toNumber);
+		}
+	}
+}
+
+window.addEventListener("resize", debounce(resizeScreenData, 500));
+
+class pageNavigation {
+	nextBtn: HTMLElement | null;
+	previousBtn: HTMLElement | null;
+	firstPageBtn: HTMLElement | null;
+	lastPageBtn: HTMLElement | null;
+	confirmationBtn: HTMLElement | null;
+
+	constructor() {
+		this.nextBtn = document.getElementById("next-records-btn");
+		this.previousBtn = document.getElementById("previous-records-btn");
+		this.firstPageBtn = document.getElementById("first-page-btn");
+		this.lastPageBtn = document.getElementById("last-page-btn");
+		this.confirmationBtn = document.getElementById("confirmation-btn");
+
+		if (this.confirmationBtn !== null) {
+			this.confirmationBtn.addEventListener("click", recordSelection);
+		}
+
+		if (this.nextBtn !== null && this.previousBtn !== null && this.firstPageBtn !== null && this.lastPageBtn !== null) {
+			let nextPage = () => {
+				let numberOfRows = Math.floor(window.innerHeight / 50);
+				fromNumber = fromNumber + numberOfRows * count;
+				let toNumber = fromNumber + numberOfRows;
+
+				let finalRecord = recordNumberTotal - 1;
+
+				(this.previousBtn as HTMLButtonElement).disabled = false;
+
+				if (toNumber >= finalRecord) {
+					(this.nextBtn as HTMLButtonElement).disabled = true;
+					fromNumber = finalRecord - numberOfRows;
+				}
+
+				getRecords(fromNumber, toNumber);
+				count = 0;
+			};
+			this.nextBtn.addEventListener("click", () => {
+				count++;
+				nextPage = debounce(nextPage, 500);
```

Comment:
If your deadset on debouncing this functionality, then you should debounce the `getRecords` function call inside `nextPage` instead, that way you can get rid of the `count` variable.